test/integration/model/upsert.test.js

```diff
@@ -248,6 +249,41 @@ describe(Support.getTestDialectTeaser('Model'), function() {
         });
       });
 
+      it('does not overwrite createdAt time on update', function() {
+        var originalCreatedAt;
+        return this.User.create({ id: 42, username: 'john'}).bind(this).then(function() {
+          return this.User.findById(42);
+        }).then(function(user) {
+          originalCreatedAt = user.createdAt;
+          return this.sequelize.Promise.delay(1000).bind(this).then(function() {
```

Comment:
In this situation I think we need the actual delay so that sequelize can internally get the proper Date() times on both create & upsert. I've used sinon to fake timers before, but iirc it has weak/no support for modifying the actual system time which is what we need here.